### PR TITLE
Add new keywords to ItemDisplay Config

### PR DESCRIPTION
**Pull request by JimmyTheJ**

### CODE REVIEW

File: BH/Modules/Item/Item.cpp [hunk 1]
```diff
@@ -42,6 +45,29 @@ void Item::LoadConfig() {
 	Toggles["Item Close Notifications"] = BH::config->ReadToggle("Item Close Notifications", "None", false);
 	Toggles["Allow Unknown Items"] = BH::config->ReadToggle("Allow Unknown Items", "None", false);
 
+	// To enable good character skills and good skill tab lists
+	Toggles["Skills"] = BH::config->ReadToggle("Skills", "None", false);
```
Comment (planqi) [inline BH/Modules/Item/Item.cpp]: Are these toggles needed? Will it work (without crashing) when `SkillsList` and `TabSkillsList` do not exist in the cfg?

File: BH/Modules/Item/ItemDisplay.cpp [hunk 2]
```diff
@@ -597,6 +646,12 @@ void Condition::BuildConditions(vector<Condition*> &conditions, string token) {
 	} else if (key.compare(0, 4, "MANA") == 0) {
 		// For unknown reasons, the game's internal Mana stat is 256 for every 1 displayed on item
 		Condition::AddOperand(conditions, new ItemStatCondition(STAT_MAXMANA, 0, operation, value * 256));
+	} else if (key.compare(0, 6, "GOODSK") == 0) {
+		Condition::AddOperand(conditions, new SkillListCondition(operation, 0, value));
```
Comment (planqi) [inline BH/Modules/Item/ItemDisplay.cpp]: The second param that is used for `type` should be a constant.

File: BH/Modules/Item/ItemDisplay.cpp [hunk 3]
```diff
@@ -1095,6 +1216,28 @@ bool ResistAllCondition::EvaluateInternalFromPacket(ItemInfo *info, Condition *a
 			IntegerCompare(pRes, operation, targetStat));
 }
 
+bool AddCondition::EvaluateInternal(UnitItemInfo *uInfo, Condition *arg1, Condition *arg2) {
+	int value = 0;
+	vector<string> codes = split(key, '+');
```
Comment (planqi) [inline BH/Modules/Item/ItemDisplay.cpp]: Please don't do string splits/compares in an evaluation method - this gets called *a lot*. Do any heavy lifting/string parsing etc when you create the `AddCondition` object.

File: BH/Modules/Item/ItemDisplay.cpp [hunk 4]
```diff
@@ -1095,6 +1216,28 @@ bool ResistAllCondition::EvaluateInternalFromPacket(ItemInfo *info, Condition *a
...
+	for (unsigned int i = 0; i < codes.size(); i++) {
```
Comment (planqi) [inline BH/Modules/Item/ItemDisplay.cpp]: These loops should be replaced with something like this (pseudo-ish code...) 
```
for each stat in stats
  int tmpVal = D2COMMON_GetUnitStat(uInfo->item, stat, 0);
  if (stat == STAT_MAXHP || stat == STAT_MAXMANA) {
    tmpVal /= 256;
  }
  value += tmpVal;
```

File: BH/Modules/Item/ItemDisplay.cpp [hunk 5]
```diff
@@ -1,4 +1,28 @@
 #include "ItemDisplay.h"
+#include "Item.h"
+
+const int COMBO_LIST_SIZE = 8;
+const char * COMBO_STAT_KEY[COMBO_LIST_SIZE]{
```
Comment (planqi) [inline BH/Modules/Item/ItemDisplay.cpp]: This is really a dictionary lookup as it is a key/value lookup.

File: BH/Modules/Item/ItemDisplay.cpp [hunk 6]
```diff
@@ -1003,6 +1057,73 @@ bool EDCondition::EvaluateInternalFromPacket(ItemInfo *info, Condition *arg1, Co
...
+		for (unsigned int i = 0; i < goodTabSkills.size(); i++) {
+			value += D2COMMON_GetUnitStat(uInfo->item, STAT_SKILLTAB, goodTabSkills.at(i));
+		}
+	}
+
+	return IntegerCompare(value, operation, targetStat);
+}
+
+bool SkillListCondition::EvaluateInternalFromPacket(ItemInfo *info, Condition *arg1, Condition *arg2) {
+	// Unecessary ?
```
Comment (planqi) [inline BH/Modules/Item/ItemDisplay.cpp]: This might only effect item drop notifications.

File: BH/Modules/Item/Item.cpp (hunk 1 above)
Comment (JimmyTheJ) [reply BH/Modules/Item/Item.cpp]: They are not needed. It does not crash when they are not there.
The GOODSK and GOODTBSK just won't work, exactly as if you have them set to false.

File: BH/Modules/Item/ItemDisplay.cpp (hunk 2 above)
Comment (JimmyTheJ) [reply BH/Modules/Item/ItemDisplay.cpp]: What do you mean ?

Comment (planqi) [reply BH/Modules/Item/ItemDisplay.cpp]: The code should be called something like this: `SkillListCondition(operation, CLASS_SKILLS, value)` and `SkillListCondition(operation, CLASS_TAB_SKILLS, value)` then later on in the evaluate function you should test against it like this: `if (type == CLASS_SKILLS)`

Pick constant names that make sense are match other conventions used, those might not be the best ones. The goal here is to make the code more human readable. Assign them in Constants.h

File: BH/Modules/Item/Item.cpp (hunk 1 above)
Comment (planqi) [reply BH/Modules/Item/Item.cpp]: BTW, the convention for reading config values changed with McGod's PR. This won't work as is now.

**Pull request merged**